### DISCUSSION TITLE
Keda Cloudwatch scaler retrieve AWS ALB target group dimension value automatically

### DISCUSSION THREAD

**Discussion by abhiptl** [kedacore/keda#4888] [Q&A / Need Help]
### Proposal

AWS Cloudwatch scaler has to be provided with a target group dimension value. If you use the AWS ALB controller then ALB and also target group are created by the AWS ALB controller for the services. This means you have to explicitly hard code the value of the target group value. If you are using argoCD then it has to be hard coded or variable. 
hardcode

**dimensionValue: "targetgroup/k8s-dla2-legacypr-99690e15ee/59179bb1028387a7"**

### Use-Case

ScaledObject already has deployment information as scaleTargetRef. If we make this dimension value option then it should automatically retrieve the value of targetGroup value based on deployment -> service and AWS ALB controller configuration.

### Is this a feature you are interested in implementing yourself?

No

**Comment by JorTurFer** [comment]
Hello,
I don't get your point (probably because I don't have knowledge about CloudWatch). So, your dimensionValue is `"targetgroup/k8s-dla2-legacypr-99690e15ee/59179bb1028387a7"` and you need to hardcode it, but you say that the deployment has that info. Is that info provided to the workload as environment variable or similar? How does the deployment know that info?

Currently we don't support reading that parameter from other source than the trigger metadata but it's just because we haven't needed it yet. Could reading the value from an environment variable of the workload solve your issue?

**Reply by vgrigoruk** [reply]
no, this value is not present neither in deployment metadata, nor as env variable. It is only present in `TargetGroupBinding` CR, which AWS load balancer controller creates based on `Service` definition:

```
Name:         k8s-namespace-service-47c684008a
Namespace:    namespace
Labels:       ingress.k8s.aws/stack=lbname
Annotations:  <none>
API Version:  elbv2.k8s.aws/v1beta1
Kind:         TargetGroupBinding
Metadata:
  Creation Timestamp:  2024-11-07T11:08:02Z
  Finalizers:
    elbv2.k8s.aws/resources
  Generation:        1
  Resource Version:  415463726
  UID:               6b792675-0968-4d65-8bc2-fdb3d69381c9
Spec:
  Ip Address Type:  ipv4
  Networking:
    Ingress:
      From:
        Security Group:
          Group ID:  sg-XXXXXXXXX
      Ports:
        Port:      http
        Protocol:  TCP
  Service Ref:
    Name:            service
    Port:            http
  Target Group ARN:  arn:aws:elasticloadbalancing:eu-west-1:XXXXXXXXX:targetgroup/k8s-namespace-service-47c684008a/3233b78e97a178bb
  Target Type:       ip
  Vpc ID:            vpc-XXXXXXXX
```
